astrolabe/spec_runner.py

```diff
@@ -213,33 +210,40 @@ def run(self, persist_cluster=False):
         # Step-5: interrupt driver workload and capture streams
         LOGGER.info("Stopping workload executor [PID: {}]".format(
             worker_subprocess.pid))
-        os.kill(worker_subprocess.pid, self.sigint)
-        stdout, stderr = worker_subprocess.communicate()
-        LOGGER.info("Stopped workload executor")
+        os.killpg(worker_subprocess.pid, self.sigint)
```

Comment:
I know this was not added in this PR, but I believe the standard signal for termination is sigterm, not sigint. The two are often handled similarly but technically they are different signals and especially daemons are not always expecting sigint (since they are not interactive).

Reply:
Agreed. The reason SIGINT was chosen here instead of SIGTERM is to allow us to be able to differentiate between the termination of the workload executor by the test framework and the termination of the workload executor due to other reasons. SIGINT is always expected to be issued by the user whereas SIGTERM is not. 

Also, I don't believe it is possible to catch SIGTERM and do any special handling for it (which we need to do in order to quit cleanly when the framework requests the workload to stop running and no errors have been encountered). To quote the UNIX wiki: 
> The SIGKILL signal is sent to a process to cause it to terminate immediately (kill). In contrast to SIGTERM and SIGINT, this signal cannot be caught or ignored, and the receiving process cannot perform any clean-up upon receiving this signal.